components/radio-button/script.js

```diff
@@ -15,141 +15,136 @@ class GamefaceRadioGroup extends HTMLElement {
 		super();
 
 		this.template = template;
+		this.previouslyCheckedElement = null;
 	}
 
 	get allButtons() {
 		return Array.from(this.querySelectorAll('radio-button'));
 	}
 
-	connectedCallback() {
-		components.loadResource(this)
-			.then((result) => {
-				this.template = result.template;
-				components.renderOnce(this);
+	setCheckedToPreviousItem(button) {
+		const prevSibling = button.previousElementSibling;
 
-				this.firstElementChild.setAttribute('role', 'radiogroup');
-			})
-			.catch(err => console.error(err));
+		if (!prevSibling) return
+		button.previousElementSibling.checked = true;
+		button.previousElementSibling.focus();
 	}
-}
 
-class RadioButton extends HTMLElement {
-	constructor() {
-		super();
-	}
+	setCheckedToNextItem(button) {
+		const nextSibling = button.nextElementSibling;
 
-	get checked() {
-		return this.getAttribute('aria-checked');
+		if (!nextSibling) return;
+		button.nextElementSibling.checked = true;
+		button.nextElementSibling.focus();
 	}
 
-	set checked(value) {
-		for (const button of this.allButtons) {
-			if (button === this) continue;
-
-			button.setAttribute('aria-checked', 'false');
-			button.setAttribute('tabindex', '-1');
-		}
+	// "this" is referencing the radio button element.
+	handleClick() {
+		this.checked = true;
+		this.focus();
+	}
 
-		if (value) {
-			this.setAttribute('tabindex','0');
-		} else {
-			this.setAttribute('tabindex','-1');
+	// "this" is referencing the radio button element.
+	handleKeydown(event) {
+		switch (event.keyCode) {
+			case KEYCODES.UP:
+			case KEYCODES.LEFT:
+				this.radioGroup.setCheckedToPreviousItem(this);
+				break;
+			case KEYCODES.DOWN:
+			case KEYCODES.RIGHT:
+				this.radioGroup.setCheckedToNextItem(this);
+				break;
+			default:
+				break;
 		}
+	};
 
-		this.setAttribute('aria-checked', value);
+	setButtonRoleAttribute(button) {
+		button.setAttribute('role', 'radio')

```

Comment:
Is there any problem to directly set this attribute to the radio element inside the radio-button template and not when the radio group connectedCallback is called via javascript?